pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -815,13 +865,35 @@ func TestSufficientCapacityWithTerminatedPodsDaemonLaunchesPod(t *testing.T) {
 		manager.dsStore.Add(ds)
 		syncAndValidateDaemonSets(t, manager, ds, podControl, 1, 0, 1)
 	}
+
+	{
+		strategy := newRollbackStrategy()
+		podSpec := resourcePodSpec("too-much-mem", "75M", "75m")
+		ds := newDaemonSet("foo")
+		ds.Spec.UpdateStrategy = *strategy
+		ds.Spec.Template.Spec = podSpec
+		manager, podControl, _, err := newTestController(ds)
+		if err != nil {
+			t.Fatalf("error creating DaemonSets controller: %v", err)
+		}
+		node := newNode("too-much-mem", nil)
+		node.Status.Allocatable = allocatableResources("100M", "200m")
+		manager.nodeStore.Add(node)
+		manager.podStore.Add(&v1.Pod{
+			Spec:   podSpec,
+			Status: v1.PodStatus{Phase: v1.PodSucceeded},
+		})
+		manager.dsStore.Add(ds)
+		syncAndValidateDaemonSets(t, manager, ds, podControl, 1, 0, 2)
```

Comment:
Why 2 events? Is that the only difference? Define a func that captures everything else e.g.
```
validate := func(strategy, expectedEvents) {
...

validate(strategy1, 1)
validate(strategy2, 2)
```